float64 x
float64 y
float64 radius
float64 points
bool loop
---
bool success
